textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/numbermatcher/NumberMatcherTest.java

```diff
@@ -0,0 +1,357 @@
+package edu.uci.ics.textdb.dataflow.numbermatcher;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.storage.IDataStore;
+import edu.uci.ics.textdb.api.storage.IDataWriter;
+import edu.uci.ics.textdb.common.constants.DataConstants;
+import edu.uci.ics.textdb.common.constants.DataConstants.NumberMatchingType;
+import edu.uci.ics.textdb.common.constants.TestConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.*;
+import edu.uci.ics.textdb.dataflow.source.ScanBasedSourceOperator;
+import edu.uci.ics.textdb.dataflow.utils.TestUtils;
+import edu.uci.ics.textdb.storage.DataReaderPredicate;
+import edu.uci.ics.textdb.storage.DataStore;
+import edu.uci.ics.textdb.storage.reader.DataReader;
+import edu.uci.ics.textdb.storage.writer.DataWriter;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by sweetest.sj on 10/4/16.
+ */
+public class NumberMatcherTest {
+
+    private IDataWriter dataWriter;
+    private DataStore dataStore;
+    private Analyzer analyzer;
+
+    private ScanBasedSourceOperator getScanSourceOperator(IDataStore dataStore) throws DataFlowException {
+        DataReaderPredicate dataReaderPredicate = new DataReaderPredicate(
+                new MatchAllDocsQuery(), DataConstants.SCAN_QUERY, dataStore,
+                dataStore.getSchema().getAttributes(), analyzer);
+        DataReader dataReader = new DataReader(dataReaderPredicate);
+        ScanBasedSourceOperator scanSource = new ScanBasedSourceOperator(dataReader);
+        return scanSource;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        dataStore = new DataStore(DataConstants.INDEX_DIR, TestConstants.SCHEMA_PEOPLE);
+        analyzer = new StandardAnalyzer();
+        dataWriter = new DataWriter(dataStore, analyzer);
+        dataWriter.clearData();
+        dataWriter.writeData(TestConstants.getSamplePeopleTuples());
+    }
+
+    @After
+    public void cleanUp() throws Exception {
+        dataWriter.clearData();
+    }
+
+    public List<ITuple> getDoubleQueryResults(double threshold, Attribute attribute, NumberMatchingType operatorType)
+            throws DataFlowException {
+        // Perform the query
+        NumberPredicate numberPredicate = new NumberPredicate(threshold, attribute, operatorType);
+        return getNumberQueryResults(numberPredicate);
+    }
+
+    public List<ITuple> getIntegerQueryResults(double threshold, Attribute attribute, NumberMatchingType operatorType)
```

Comment:
it the threshold here `int` ? `getIntegerQueryResults` is exactly the same as `getDoubleQueryResults`.